packages/frontend/src/pages/Contribute.jsx

```diff
@@ -21,7 +26,40 @@ const ContributeState = {
   offline: 5,
 }
 
+const vertexShaderSrc = `
+  varying vec3 vPosition;
+  varying float vMass;
+
+  attribute float mass;
+
+  void main() {
+    vPosition = position;
+    vMass = mass;
+
+    gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.);
+    gl_PointSize = max(1., vMass);
+  }
+`
+
+const fragmentShaderSrc = `
+varying vec3 vPosition;
+varying float vMass;
+
+void main() {
+    vec3 color = vMass * vec3(236., 46., 0.) + (1. - vMass) * vec3(1., 94., 158.);
+    if (vMass > 1.) color = vec3(255., 255., 0.);
+    gl_FragColor = vec4(color / 255., 1.0);
```

Comment:
I think here could also be random according to user's `ceremony.authToken`, then it would make everyone's galaxy looks a lot more different.

Reply:
This part is used for assigning colors. `vec3 color = vMass * vec3(236., 46., 0.) + (1. - vMass) * vec3(1., 94., 158.);` effectively is 
```js
if vMass == 1.0 {
color = vec3(236, 46, 0)
}
else  if vMass == 0.0 {
color = vec3(1, 94, 158)
}
```.
The galaxies are already initialized with a specific amount of randomness such that they don't "explode" in the first 0.1s. I thought it would make more sense for users to seed randomness by adding particles. The other thing I can do is automatically add a few "user particles (the yellow ones)" which are several orders of magnitude greater in mass than the stellar/halo particles.